kiwi_lint/generic_foreign_key.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+from pylint.checkers import utils, BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint_django.utils import node_is_subclass
+
+
+class GenericForeignKeyChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+
+    name = 'generic-foreign-key-checker'
+
+    msgs = {
+        'R4525': (
+            'Avoid using GenericForeignKey',
+            'avoid-generic-foreign-key',
+            'Avoid using django.contrib.contenttypes.fields.GenericForeignKey',
+        )
+    }
+
+    @utils.check_messages('avoid-generic-foreign-key')
+    def visit_classdef(self, node):
+
+        if not node_is_subclass(node, 'django.db.models.base.Model'):
+            return
+
+        for part in node.body:
+            if not isinstance(part, astroid.Assign):
+                continue
+
+            if not isinstance(part.value, astroid.Call):
+                continue
+
+            if 'GenericForeignKey' in part.value.func.as_string():
```

Comment:
this check should be more robust. Try using node_is_subclass here too.

Reply:
I am not sure I understand correctly.
If I pass `list(part.value.func.infer())[0]` to node_is_subclass it sees it as `django.db.models.fields.mixins.FieldCacheMixin`

Reply:
IMO we should first figure out if this implementation stays like it is or will change with different visit_ methods and then figure out how to appropriately assert if something is a GenericForeignKey or not.  

Maybe infer() doesn't give the expected result, IDK.

Reply:
I will try to find some spare time to check astroid's documentation and find a better way to do this.
In the meantime If anyone wants to take a shot for the bounty, feel free to use any/all parts of my code.